fix: Ignore Tableau dashboards of insufficient privileges

shuichiro-makigaki opened the pull request:
### Summary of Changes

Closes: amundsen-io/amundsen#911

Tableau API returns null for dashboard name or project name when the user token does not have access privilege for the dashboard or the project.
The sanitizers in `TableauDashboardUtils` will raise `TypeError` when the value is None.
If these names are None, we should ignore them and shows some warnings.

### Tests

No feature added.

### CheckList

Make sure you have checked **all** steps below to ensure a timely review.

- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes.
- [x] PR adds unit tests, updates existing unit tests, **OR** documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
- [x] PR passes `make test`

dorianj reviewed (approved):
LGTM -- would it be difficult to add a test for this behavior?

Also, please re-push to get DCO

feng-tao commented:
flake8 fails:
```

========================= 303 passed in 43.64 seconds ==========================
flake8 .
/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/pycodestyle.py:113: FutureWarning: Possible nested set at position 1
  EXTRANEOUS_WHITESPACE_REGEX = re.compile(r'[[({] | []}),;:]')
./databuilder/extractor/dashboard/tableau/tableau_dashboard_last_modified_extractor.py:43: [E501] line too long (170 > 120 characters)
./databuilder/extractor/dashboard/tableau/tableau_dashboard_extractor.py:44: [E501] line too long (170 > 120 characters)
Makefile:11: recipe for target 'lint' failed
make: *** [lint] Error 1
```

shuichiro-makigaki commented:
@dorianj 
> LGTM -- would it be difficult to add a test for this behavior?

No. I added some tests.

> Also, please re-push to get DCO

done!

@feng-tao 
> flake8 fails:

fixed!

dorianj commented:
thank you for your contribution, @shuichiro-makigaki !
